presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java

```diff
@@ -69,15 +78,23 @@ private SpatialJoinUtils() {}
     private static boolean isSupportedSpatialFunction(FunctionCall functionCall)
     {
         String functionName = functionCall.getName().getSuffix();
-        return functionName.equalsIgnoreCase(ST_CONTAINS.getSuffix()) || functionName.equalsIgnoreCase(ST_WITHIN.getSuffix())
-                || functionName.equalsIgnoreCase(ST_INTERSECTS.getSuffix());
+        for (FullyQualifiedName allowedName : ALLOWED_SPATIAL_JOIN_FUNCTIONS) {
```

Comment:
this loop is slow; why not convert ALLOWED_SPATIAL_JOIN_FUNCTIONS into a set of  lowercase function names, then use `Set.contains(functionName.toLower())`